docs: budget draft (#310)
blockfrost/blockfrost-platform: docs/src/content/budget.mdx

### Audit

Following the completion of the Icebreaker and Hopper eras, we plan to conduct a security audit of the developed code by a trusted third party. We estimate the cost of this audit to be $50,000.

## Community Engagement

As part of the budget process, we plan to host several recorded community calls for DReps and other community members to discuss this proposal.

Records of these calls will be posted below.

If you have questions that is not addressed in our FAQ or covered in this document, feel free to reach out to the team through our support channels:

* Discord: [IOG's technical Discord](https://discord.gg/3dHCe6X88f)
* E-mail: support@blockfrost.io
* Twitter/X: [@Blockfrost_io](https://x.com/Blockfrost_io)

## FAQ

We have collected questions from the community and listed them here.

* Can I also contribute to these efforts?
  * Definitely! All the deliverables are open-source and we welcome everyone who want to contribute to open an issue on our Github.
* Didn't IOG already post their budget? Why you are not including there?
  * IOE (Input Output Engineering) is a different entity than Blockfrost, therefore we are proposing our own budget.

> 

Author: ```suggestion
* Is there any dependency or overlap between this proposal and past Catalyst proposals?
  * There is no dependency or overlap of this proposal with any other Blockfrost proposal. However, there are dependencies that we're using that has been funded by project Catalyst in the future. Pallas is a Rust library that I know have received funding before. We are also going to use Cardano Data nodes such as Dolos that I also know have received funding before too. We are building on shoulders of giants and none of this would be possible without these mature tools.
* Following completion of this proposal, will there be any additional requests to Treasury or Catalyst submissions related to the maintenance of this proposal deliverables, or will this service be self-sustaining once work on this proposal is complete?
  * At this moment, we do not know. As with all software in use, the Blockfrost Platform will also require resources for maintenance. That said, we're in the early stages and are exploring potential other sources of funding. As we are slowly handing over revenue to SPOs, we were thinking perhaps a DAO model, where a portion of the fees goes to the "Blockfrost treasury," could fund this in the future. Full disclosure, we have also discussed a potential token launch that might be used to secure this DAO. However, this is still early and needs discussion with the community too.
* Can you show in some way that 1.3 million ada is a reasonable cost? For example, can you get recommendations from other community developers that this is roughly what it should be, or provide an estimate of the amount of work required?
  * This is an excellent question, and I like the idea of gathering insights from other builderrs in the community. I'm not certain if this is the optimal approach, but we have posted a call for opinions on our Twitter, and I hope we will receive valuable feedback on this topic. The discussion is happening at [this thread](https://x.com/blockfrost_io/status/1913567785659023740).
```

#### Hopper era

* [Hopper: Research into payments channels between the platform and the consumer](https://github.com/blockfrost/blockfrost-platform/issues/287)
* [Hopper: Hydra payment channels between platform and gateway](https://github.com/blockfrost/blockfrost-platform/issues/277)
* [Hopper: Hydra payment channel between gateway and consumer](https://github.com/blockfrost/blockfrost-platform/issues/278)

## Position with the Cardano Vision and Roadmap for 2025

In March, the majority of DReps voted to adopt the strategic document [Cardano Vision and Roadmap for 2025](https://adastat.net/governances/56f39054758f1a3cedc1de9225d66bf270b62dfdbfbc5399f1d6d43aceffc63600).

Our proposal directly addresses the document's item:

  "Decentralize data API services to reduce reliance on centralized providers and empower SPOs."
  
We are decentralizing the Blockfrost API and increasing the SPO incentives, at the same time.

## Proposed budget

The majority of our budget is allocated to developer salaries. We are requesting an annual budget of $120,000  to support one full-time employee (FTE).

> Our total funding request from the treasury is $650,000, which is equivalent to approximately 988,708 ADA based on the current rate of 0.65 USD per ADA.

Author: ```suggestion
Our total funding request from the treasury is $650,000, which is equivalent to approximately 1,300,000 ADA based on the current rate of 0.50 USD per ADA.
```

Looks like most of the proposals are counting with half a dollar per ADA as a conservative estimate.

### Team

The team that is working on the deliverble is as follows:

* [Marek](https://github.com/mmahut): Primary project lead (part-time)
* [Beatrice](https://github.com/opacular): Community engagement lead and onboarding (part-time)
* [Vladimir](https://github.com/vladimirvolek): Gateway lead
* [Michal](https://github.com/michalrus): Platform lead
* [Sefa](https://github.com/ginnun): Caching lead
* [Bart](https://github.com/bslabiak): Conformance testing lead

This amounts to total of 5 FTEs.

### Audit

Following the completion of the Icebreaker and Hopper eras, we plan to conduct a security audit of the developed code by a trusted third party. We estimate the cost of this audit to be $50,000.

> 

Author: ```suggestion
### Office pet plant

$50 of the budget will be allocated for acquisition and maintenance of Rosa canina plant in our home offices, also known as the Dog Rose, which we will name Hosky. Bi-weekly proof of life will be provided on our social media.
```

* Following completion of this proposal, will there be any additional requests to Treasury or Catalyst submissions related to the maintenance of this proposal deliverables, or will this service be self-sustaining once work on this proposal is complete?
  * At this moment, we do not know. As with all software in use, the Blockfrost Platform will also require resources for maintenance. That said, we're in the early stages and are exploring potential other sources of funding. As we are slowly handing over revenue to SPOs, we were thinking perhaps a DAO model, where a portion of the fees goes to the "Blockfrost treasury," could fund this in the future. Full disclosure, we have also discussed a potential token launch that might be used to secure this DAO. However, this is still early and needs discussion with the community too.
* Can you show in some way that 1.3 million ada is a reasonable cost? For example, can you get recommendations from other community developers that this is roughly what it should be, or provide an estimate of the amount of work required?

>   * This is an excellent question, and I like the idea of gathering insights from other builderrs in the community. I'm not certain if this is the optimal approach, but we have posted a call for opinions on our Twitter, and I hope we will receive valuable feedback on this topic. The discussion is happening at [this thread](https://x.com/blockfrost_io/status/1913567785659023740).

Author: ```suggestion
  * This is an excellent question, and I like the idea of gathering insights from other builderrs in the community. I'm not certain if this is the optimal approach, but we have posted a call for opinions on our Twitter, and I hope we will receive valuable feedback on this topic. The discussion is happening at [this thread](https://x.com/blockfrost_io/status/1913567785659023740).
* Will anyone be able to run an Icebreaker, or does this needs approval from Blockfrost? Will there be a limit on the number of Icebreakers?  
  * At the moment, this is limited by a NFT license (that we provide for free, we just want to have contact information of which SPO is running which node, all one need is just to fill up a form to get one), given we're just in beta testing phase. When the beta testing is over, anyone will be able to join the clusters - not only SPOs but also full wallet users. Additionally, I think an important note is that anyone would be able to run their entire Blockfrost gateway cluster and "their own" Icebreakers. I envision that groups of SPOs might get together and start their own clusters with their own pricing, SLAs, endpoints etc and start to compete. We are also researching how we could have a decentralized P2P marketplace for Icebreakers to offer their services (imagine just setting the price per request on your Icebreaker, starting it and it will just connect to a cluster without you having to choose one) - but this is out of scope for this year as much more research needs to be done.

```

>   - The pricing will be per request. When you look at our website, the price customer pays for 1M requests is around ~4 ADA. At the moment, we are rewarding Icebreakers with a minimum of $20 in ADA monthly, as the traffic to them is still small (under $20), but as soon that traffic increases, they will be paid what they have served. In the future, when the full traffic is going to be handled fully in decentralized way, we will introduce a tax parameter, that will take a cut of all revenue and allocate it to the Blockfrost gateway to fund the infrastructure and administrative fees for running it. I imagine people will compete in a way that will start new Gateways with lower taxes to attract the Icebreakers. We are also investigating a possible DAO for this, but this is still in research and out of scope for this 2025 proposal.

Author: ```suggestion
  - The pricing will be per request. When you look at our website, the price customer pays for 1M requests is around ~4 ADA. At the moment, we are rewarding Icebreakers with a minimum of 20 USD in ADA monthly, as the traffic to them is still small (under 20 USD), but as soon that traffic increases, they will be paid what they have served. In the future, when the full traffic is going to be handled fully in decentralized way, we will introduce a tax parameter, that will take a cut of all revenue and allocate it to the Blockfrost gateway to fund the infrastructure and administrative fees for running it. I imagine people will compete in a way that will start new Gateways with lower taxes to attract the Icebreakers. We are also investigating a possible DAO for this, but this is still in research and out of scope for this 2025 proposal.
```

* Will anyone be able to run an Icebreaker, or does this needs approval from Blockfrost? Will there be a limit on the number of Icebreakers?  
  * At the moment, this is limited by a NFT license (that we provide for free, we just want to have contact information of which SPO is running which node, all one need is just to fill up a form to get one), given we're just in beta testing phase. When the beta testing is over, anyone will be able to join the clusters - not only SPOs but also full wallet users. Additionally, I think an important note is that anyone would be able to run their entire Blockfrost gateway cluster and "their own" Icebreakers. I envision that groups of SPOs might get together and start their own clusters with their own pricing, SLAs, endpoints etc and start to compete. We are also researching how we could have a decentralized P2P marketplace for Icebreakers to offer their services (imagine just setting the price per request on your Icebreaker, starting it and it will just connect to a cluster without you having to choose one) - but this is out of scope for this year as much more research needs to be done.

> 

Author: ```suggestion
* How will the revenue sharing model for Icebreakers be structured? Specifically, could you elaborate on how the share of Blockfrost revenue will be calculated and distributed to participating node operators?
  * The pricing will be per request. When you look at our website, the price customer pays for 1M requests is around ~4 ADA. At the moment, we are rewarding Icebreakers with a minimum of $20 in ADA monthly, as the traffic to them is still small (under $20), but as soon that traffic increases, they will be paid what they have served. In the future, when the full traffic is going to be handled fully in decentralized way, we will introduce a tax parameter, that will take a cut of all revenue and allocate it to the Blockfrost gateway to fund the infrastructure and administrative fees for running it. I imagine people will compete in a way that will start new Gateways with lower taxes to attract the Icebreakers. We are also investigating a possible DAO for this, but this is still in research and out of scope for this 2025 proposal.

```
